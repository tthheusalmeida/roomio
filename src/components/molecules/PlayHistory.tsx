export interface Move {
  x: number;
  y: number;
  player: "x" | "o";
}

interface PlayHistoryProps {
  history: Move[];
}

export default function PlayHistory({ history }: PlayHistoryProps) {
  return (
    <div className="bg-gray-800 rounded-md p-4">
      <h3 className="font-semibold text-xs mb-2 text-gray-200 z-20 relative">
        History:
      </h3>

      <div className="h-36 max-h-36 overflow-y-auto relative">
        <ul className="flex flex-col gap-1 text-sm text-gray-400 relative z-0">
          {[...history].reverse().map((move, index) => (
            <li
              className={`list-none ${
                index === 0 ? "font-semibold text-white" : ""
              }`}
              key={index}
            >
              {move.player.toUpperCase()} played at ({move.x + 1}, {move.y + 1})
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
}
