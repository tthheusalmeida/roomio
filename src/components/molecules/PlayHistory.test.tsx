import { render, screen } from "@testing-library/react";
import PlayHistory, { Move } from "./PlayHistory";

describe("PlayHistory component", () => {
  const history: Move[] = [
    { x: 0, y: 0, player: "x" },
    { x: 1, y: 1, player: "o" },
    { x: 2, y: 2, player: "x" },
  ];

  it("renders the title", () => {
    render(<PlayHistory history={[]} />);
    expect(screen.getByText("History:")).toBeInTheDocument();
  });

  it("renders moves in reverse order", () => {
    render(<PlayHistory history={history} />);
    const items = screen.getAllByRole("listitem");
    expect(items.length).toBe(3);

    expect(items[0].textContent).toContain("X played at (3, 3)");
    expect(items[1].textContent).toContain("O played at (2, 2)");
    expect(items[2].textContent).toContain("X played at (1, 1)");
  });

  it("applies highlight class to the last move", () => {
    render(<PlayHistory history={history} />);
    const items = screen.getAllByRole("listitem");
    expect(items[0]).toHaveClass("font-semibold");
    expect(items[0]).toHaveClass("text-white");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<PlayHistory history={history} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
