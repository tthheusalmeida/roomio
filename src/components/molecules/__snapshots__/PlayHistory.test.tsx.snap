// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PlayHistory component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-800 rounded-md p-4"
  >
    <h3
      class="font-semibold text-xs mb-2 text-gray-200 z-20 relative"
    >
      History:
    </h3>
    <div
      class="h-36 max-h-36 overflow-y-auto relative"
    >
      <ul
        class="flex flex-col gap-1 text-sm text-gray-400 relative z-0"
      >
        <li
          class="list-none font-semibold text-white"
        >
          X played at (3, 3)
        </li>
        <li
          class="list-none "
        >
          O played at (2, 2)
        </li>
        <li
          class="list-none "
        >
          X played at (1, 1)
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
